src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralOptions.cs

```diff
@@ -8,10 +8,9 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.SplitStringLiteral
 {
     internal class SplitStringLiteralOptions
     {
-        public const string FeatureName = "SplitStringLiteral";
-
         public static PerLanguageOption<bool> Enabled =
-            new PerLanguageOption<bool>(FeatureName, nameof(Enabled), defaultValue: true);
+            new PerLanguageOption<bool>(nameof(SplitStringLiteralOptions), nameof(Enabled), defaultValue: true,
+                serializations: new RoamingProfileSerialization("TextEditor.%LANGUAGE%.Specific.SplitStringLiterals"));
```

Comment:
The goal is to save the setting with the same name as the old code did. Since settings roam between Visual Studio 2015 and "15", we want to use the same names as before, which was TextEditor.CSharp/Basic.Specific.Blah. I decided to be very explicit here and "unhide" the qualification, which was previously completely buried away inside the serializer. You'll see cases if you go through all of these where I'm using "strange" names to maintain compability where I can so things synchronize with each other nicely. That's also why I'm not using nameof: the idea is once that name is there, it's really there and should probably stay the same, even if we rename the code symbols for the options themselves.


Since these options are per-language, the %LANGUAGE% is replaced by the serializer to CSharp/Basic. I did offer an overload of RoamingProfileSerialization that instead gives you a lambda so you can control that more specifically, but the 99% case was just to use this pattern. I can't say I like it, but it seems practical.

Curiously, this one case you commented on was a headache I did change the behavior of: the old serializer would do .Specific.Blah where Blah was the name of the option. So we were serializing this to TextEditor.CSharp.Specific.Enabled! I renamed this one since the only thing I'm breaking setting sync with is a pre-release build.